headers.go

```diff
@@ -25,3 +26,24 @@ func AddHeaders(headers []string) func(http.Handler) http.Handler {
 	}
 	return fn
 }
+
+func getSiteURL(r *http.Request) string {
+	if Config.siteURL != "" {
+		return Config.siteURL
+	} else {
+		u := &url.URL{}
+		u.Host = r.Host
+
+		if Config.sitePath != "" {
+			u.Path = Config.sitePath
+		}
+
+		if scheme := r.Header.Get("X-Forwarded-Proto"); scheme != "" {
```

Comment:
Does this account for using the built-in server(s)? For example, when using tls with the built-in server, we already know that the scheme is https